opencog/rule-engine/Rule.h

```diff
@@ -106,6 +107,16 @@ class Rule : public boost::totally_ordered<Rule>
 	void init(const Handle& rule_member);
 	void init(const Handle& rule_alias, const Handle& rbs);
 	void init(const Handle& rule_alias, const Handle& rule, const Handle& rbs);
+
+    /**
+     * Verify if this rule is defined in the required format.
+     * The main purpose is to give user the corresponding warnings and info.
+     * Only verify a normal rule, do not use this function to verify a meta rule.
+     * Current verify items:
+     * 1.If there are multiple conslusions, they also need to be wrapped with one
+     *   single ExecutionOutPutLink, otherwise it won't work for bc.
+     */
+    bool verify_rule();
```

Comment:
Note that this file uses smart tabs, not whitespaces for indentation.

I really like smart tabs, it allows the user to render its code with its chosen tab size, yet while preserving the alignments no matter what, and in the end makes browsing the code slightly faster. The inconvenience is that it requires the developer have an editor that supports it. Maybe the following Stackoverflow issue can help you add smart tabs to VSCode https://stackoverflow.com/questions/2728658/are-there-any-visual-studio-add-ins-for-true-smart-tabs otherwise just use plain tabs.

Reply:
Are you sure want to make "using mart tables" as an Opencog coding rule even in c++? 
I personally fine, but everyone has its own favorite tools. I use Qt Creator. This rule may be an unattractive rules for new people to join the development of Opencog.

Reply:
What about we agree on "using smart tabs" in Scheme, and free to use whatever everyone gets used to in c++?

Reply:
Sure, I don't want to force people to use smart tabs, I've edited the dev standard to make it clearer

https://wiki.opencog.org/w/Development_standards#Formatting

basically the most important is to respect the indentation style of the file, and if the style is smart tabs, then regular tabs can be used, no big deal.

Reply:
Ok, understood.

Reply:
smart tabs or not, this is just badly indented.    The indentation used should, at a minimum, be consistent with the indentation in the rest of the file. This is not.